programs/move_loader_api/src/lib.rs

```diff
@@ -180,22 +211,24 @@ pub fn process_instruction(
                 }
                 data_store.apply_write_set(&output.write_set());
 
-                // Dump Libra account data back into Solana account data
+                // Break data store into a list of address keyed WriteSets
                 let mut write_sets = data_store.into_write_sets();
-                for (i, keyed_account) in keyed_accounts[GENESIS_INDEX..].iter_mut().enumerate() {
-                    let address = if i == 0 {
-                        // TODO: Remove this special case for genesis when genesis contains real pubkey
-                        AccountAddress::default()
-                    } else {
-                        // let mut address = [0_u8; 32];
-                        // address.copy_from_slice(keyed_account.unsigned_key().as_ref());
-                        AccountAddress::new(*to_array_32(keyed_account.unsigned_key().as_ref()))
-                    };
-                    let write_set = write_sets.remove(&address).unwrap();
+
+                // Genesis account holds both mint and stdliib under address 0x0
+                let write_set = write_sets.remove(&AccountAddress::default()).unwrap();
+                keyed_accounts[GENESIS_INDEX].account.data.clear();
+                let writer =
+                    std::io::BufWriter::new(&mut keyed_accounts[GENESIS_INDEX].account.data);
+                bincode::serialize_into(writer, &LibraAccountState::Genesis(write_set)).unwrap();
+
+                // Now do the rest of the accounts
+                for keyed_account in keyed_accounts[GENESIS_INDEX + 1..].iter_mut() {
+                    let write_set = write_sets
+                        .remove(&pubkey_to_address(keyed_account.unsigned_key()))
+                        .unwrap();
                     keyed_account.account.data.clear();
                     let writer = std::io::BufWriter::new(&mut keyed_account.account.data);
```

Comment:
It'd be nice to see this stuff in a standalone function so that we can experiment with it. At a glance, it looks like you're allowing Move to grow the account's Vec.

Reply:
That's exactly what happens now, need to address this on a larger scale